src/mappings/staking.ts

```diff
@@ -208,15 +208,33 @@ export function handleStakeDelegated(event: StakeDelegated): void {
     }
   }
 
+  let isStakeBecomingActive = delegatedStake.shareAmount.isZero() && !event.params.shares.isZero()
+
   delegatedStake.stakedTokens = delegatedStake.stakedTokens.plus(event.params.tokens)
   delegatedStake.shareAmount = delegatedStake.shareAmount.plus(event.params.shares)
   delegatedStake.lastDelegatedAt = event.block.timestamp.toI32()
   delegatedStake.save()
 
+  // reload delegator to avoid edge case where we can overwrite stakesCount if stake is new
+  delegator = Delegator.load(delegatorID) as Delegator
+
+  let isDelegatorBecomingActive = delegator.activeStakesCount == 0 && isStakeBecomingActive
+
   // upgrade graph network
   let graphNetwork = GraphNetwork.load('1')
   graphNetwork.totalDelegatedTokens = graphNetwork.totalDelegatedTokens.plus(event.params.tokens)
+
```

Comment:
I think this is a bit cleaner :

```
   // DELETE isDelegatorBecomingActive

  // upgrade graph network
  let graphNetwork = GraphNetwork.load('1')
  graphNetwork.totalDelegatedTokens = graphNetwork.totalDelegatedTokens.plus(event.params.tokens)

  if(isStakeBecomingActive) {
    graphNetwork.activeDelegationCount = graphNetwork.activeDelegationCount + 1
    delegator.activeStakesCount = delegator.activeStakesCount + 1
      if( delegator.activeStakesCount == 0) {
          graphNetwork.activeDelegatorCount = graphNetwork.activeDelegatorCount + 1
     }
  }
```
Reason being is that `isDelegatorBecomingActive` requires that `isStakeBecomingActive` is already true. Thus we are checking that `isStakeBecomingActive` is true twice. In what I laid out above, we remove the duplication of this check

Reply:
yeah, that would be cleaner, although I added both as independent variables so that it was easier to read. I'm gonna fix the missing `has` on the previous comment, so I can change this too if you'd like